### DISCUSSION TITLE
Navigation Pane / Chapters / cross referencing

### DISCUSSION THREAD

**Discussion by Tydblair** [xournalpp/xournalpp#3943] [Ideas]
Hi There, 

I dont know if there is an existing feature that would serve as a workaround, Ive had a squiz on the descussion forum. But I would Kill for a way to navigate and organise long documents. 

I Use xournal++ for taking notes for class. And I would love to be able to jump to my "week 7 - Tutorial' notes quickly, when they're on page 32. A 'bookmarks' side bar view would work really well, and a custom text object that gets inserted into the page that links to the books marks feature would be amazing. 

If there is any other navigation workaround that could be used I would be forever greatful. Or if its a relativly easy plugin to make, I would pay for that feature. Like $50 australian dollerydoos, which I know is not a whole lot, but as mentioned above, I am a broke ass student.

**Comment by bhennion** [comment]
This feature has been requested already (see issues #1643 and #846), It is however not quite on the top of the pile, and unless you try and implement it yourself (you're most welcome to do so), I don't think it'll get done anytime soon.

**Reply by netlimpopo** [reply]
My noob-workaround consists of writing the headlines of important paragraphs with text instead of ink. This makes at least the headline searchable. The I use CTRL+F to find the headline or CTRL+G for jumping to he page (introducing Codewords(sorta "tags") like "WT2" also helps, as they usually don'toccur in normal text)
Also in the dev. Version the programme can actually show the navigation toolbar of a pdf document. One could create a pdf template in Word/libreoffice or whatever with headlines then export to pdf and import to xournal ++. 
This would at least provide a (non changable) basic structure.

**Reply by modernNeo** [reply]
> It is however not quite on the top of the pile,

Is this still the case and if so, what can be done to get it higher on the list?

**Reply by netlimpopo** [reply]
There is some interest in a new file format, which would allow pointing at specific objects such as headlines. Another Pull Request deals with external Hyperllinks and could possibly internal Hyperlinks in the future. It is quite hard to implement and will take time. As it is up to the developer there is no clear answer to that, as it depends on skill and time. Sorry. I think for now creating a template on a pdf basis with a treestructure is the easiest way to implement it already now. Chapter 1, chapter 2,... need to be updated with the names of the headlines in the PDF document and therefore changing the background in the xopp file.
